lib/helpers.php

```diff
@@ -7,16 +7,31 @@
 
 date_default_timezone_set('UTC');
 
+// Optional: Config file loading
+$config_file = dirname(__FILE__).'/config.php';
 if(getenv('ENV')) {
-  require(dirname(__FILE__).'/config.'.getenv('ENV').'.php');
+  $config_file = dirname(__FILE__).'/config.'.getenv('ENV').'.php';
+}
+if(file_exists($config_file) && is_executable($config_file)) {
+  require $config_file;
 } else {
-  require(dirname(__FILE__).'/config.php');
+  $initlog = make_logger('system-init');
+  $initlog->info("No executable config file '{$config_file}' falling back to ENV");
+}
+
+function get_setting($area, $name=null) {
+  return $_ENV[strtoupper("{$area}_{$name}")] ?? try_config($area, $name);
```

Comment:
Does this mean the environment variables for things like `base` need to be defined as `BASE_`? Seems like we'd want to avoid the underscore there.

Reply:
This likely needs more thought than I gave it :wink: I can adjust to not need that.

Reply:
```suggestion
  $env_name = is_null($name) ? $area : "{$area}_{$name}";
  return $_ENV[strtoupper($env_name)] ?? try_config($area, $name);
```